sgx_urts/src/file.rs

```diff
@@ -461,9 +481,17 @@ pub extern "C" fn u_opendir_ocall(error: *mut c_int, pathname: *const c_char) ->
 pub extern "C" fn u_readdir64_r_ocall(
     dirp: *mut DIR,
     entry: *mut dirent64,
-    result: *mut *mut dirent64,
+    hint: *mut c_int,
 ) -> c_int {
-    unsafe { libc::readdir64_r(dirp, entry, result) }
+    let mut result_ptr = ptr::null_mut();
+    let result = unsafe { libc::readdir64_r(dirp, entry, &mut result_ptr) };
+    assert!(!hint.is_null());
+    if !result_ptr.is_null() {
+        unsafe {
```

Comment:
recommend check `hint` inside this block and return `EOVERFLOW`(One of the values in the structure to be returned cannot be represented correctly.) if `NULL` so we can remove this `assert`.